errtracker/errtracker.go

```diff
@@ -79,6 +81,20 @@ func readMachineID() ([]byte, error) {
 	return nil, fmt.Errorf("cannot report: no suitable machine id file found")
 }
 
+// detectPartialDpkgUpdate returns true if a partial update of the snapd dpkg
+// package is detected. This is done by inspecting the apparmor profile of
+// snap-confine which is classified as a conffile and this has different update
+// semantics than that of regular files.
+//
+// This probe is here to aid in resolving the following issue:
+// https://forum.snapcraft.io/t/test-failures-with-cannot-create-lock-directory-run-snapd-lock/390/
+func detectPartialDpkgUpdate() string {
+	if osutil.FileExists(snapConfineProfileDpkgNew) {
+		return "dpkg-new file present"
```

Comment:
Maybe the md5 hash of the file instead of this string? This way we might get a clue if all files are modified in the same way and we can compare with the hash of previous deb packages etc.